src/main.js

```diff
@@ -1,2 +1,261 @@
-import './style.css'
+
+// Import axios
+import axios from "axios";
+// Create Async function Get endpoint from REST Countries API so we can load all the data.
+
+// Create a variabele to store the array from the Async function outside the scope.
+let countries = [];
+
+
+fetchCountryData();
+
+async function fetchCountryData() {
+  // Return a Promise
+  try {
+    const response = await axios.get('https://restcountries.com/v2/all?fields=name,alpha2Code,region,capital,flag,population,currencies,subregion,languages');
+    // Let variabele get updated with the array from axios.get
+    countries = response.data;
+
+    // Sort the data population
+    sortDataPopulation(countries);
+    // console.table(countries);
+    // Display the data in the app
+    contentData(countries);
+    // console.table(countries);
+    // Adds the class region so that this can have an specific color
+    regionColorizer(countries);
+
+    popupscreen(countries);
+
+
+
+    
+            
+  } catch(e) {
+      
+    // Add error message so that the user will be informed when the API-server is not responding
+    console.error(e);
+    const errorMsg = document.querySelector('.error-msg')
+    errorMsg.innerHTML = `<h3>----- No data was loaded -----</h3>`
+    return errorMsg;
+
+  }
+  
+}
+
+
+
+// ------------ Search and Filter header -------------------------
+// Searchbar with addeventlistener on input
+const searchBar = document.getElementById("search-bar");
+searchBar.addEventListener('input', (e) => {
+
+    e.preventDefault();
+    const searchString = e.target.value;
+    // console.log(e.target.value.length);
+    filterSearch(countries, searchString);
+  
+  });
+
+// The filter search with the arguments of searchstring and array
+function filterSearch(countries, searchString) {
+  const filteredCountries = countries.filter((country) => {
```

Comment:
Hier zou je nog spreading kunnen toepassen, zodat je niet overal `country.` vóór hoeft te zetten
```suggestion
  const filteredCountries = countries.filter({ name, region } => {
```

Reply:
De spreading kreeg ik niet voor elkaar. name werd niet gevonden en kreeg hierdoor niet het gewenste resultaat. Deze laat ik voor nu zoals deze is.